File: app/src/main/java/com/example/healthalarm/ViewPagerFuncations/ViewpagerFuncation.java
package com.example.healthalarm.ViewPagerFuncations;

import android.os.Handler;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.ViewPager;

import java.util.Timer;
import java.util.TimerTask;

public class ViewpagerFuncation {

    private static final float MIN_SCALE = 0.85f;
    private static final float MIN_ALPHA = 0.5f;
    final long DELAY_MS = 1000;
    final long PERIOD_MS = 4000;
    int currentPage = 0;
    Timer timer;

    public void setviewpager(ViewPager viewPager) {

        viewPager.setPageTransformer(true, new ZoomOutPageTransformer());

        final Handler handler = new Handler();
        final Runnable Update = () -> {
            if (currentPage == 7) {
                currentPage = 0;
            }
            viewPager.setCurrentItem(currentPage++, true);
        };

        timer = new Timer();
        timer.schedule(new TimerTask() {
            @Override
            public void run() {
                handler.post(Update);
            }
        }, DELAY_MS, PERIOD_MS);
    }

    private static class ZoomOutPageTransformer implements ViewPager.PageTransformer {
        @Override
        public void transformPage(@NonNull View page, float position) {

            int pageWidth = page.getWidth();
            int pageHeight = page.getHeight();

            if (position < -1) {
                page.setAlpha(0);
            } else if (position <= 1) {

                float scaleFactor = Math.max(MIN_SCALE, 1 - Math.abs(position));
                float vertMargin = pageHeight * (1 - scaleFactor) / 2;
                float horzMargin = pageWidth * (1 - scaleFactor) / 2;

                if (position < 0) {
                    page.setTranslationX(horzMargin - vertMargin / 2);
                } else {
                    page.setTranslationX(-horzMargin + vertMargin / 2);
                }
                page.setScaleX(scaleFactor);
                page.setScaleY(scaleFactor);

                page.setAlpha(MIN_ALPHA +
                        (scaleFactor - MIN_SCALE) /
                                (1 - MIN_SCALE) * (1 - MIN_ALPHA));

            } else {
                page.setAlpha(0);
            }
        }
    }

}

File: app/src/main/java/edu/apps/healthalarm/ViewPagerFuncations/ViewpagerFuncation.kt
@file:Suppress("PackageName")

package edu.apps.healthalarm.ViewPagerFuncations

import android.os.Handler
import android.view.View
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.PageTransformer
import java.util.*
import kotlin.math.abs

class ViewpagerFuncation {
    private val DELAY_MS: Long = 1000
    private val PERIOD_MS: Long = 4000
    private var currentPage = 0
    var timer: Timer? = null

    fun setviewpager(viewPager: ViewPager) {
        viewPager.setPageTransformer(true, ZoomOutPageTransformer())
        val handler = Handler()
        val Update = Runnable {
            if (currentPage == 7) {
                currentPage = 0
            }
            viewPager.setCurrentItem(currentPage++, true)
        }
        timer = Timer()
        timer!!.schedule(object : TimerTask() {
            override fun run() {
                handler.post(Update)
            }
        }, DELAY_MS, PERIOD_MS)
    }

    private class ZoomOutPageTransformer : PageTransformer {
        override fun transformPage(page: View, position: Float) {
            val pageWidth = page.width
            val pageHeight = page.height

            when {
                position < -1 -> {
                    page.alpha = 0f
                }
                position <= 1 -> {
                    val scaleFactor = Math.max(MIN_SCALE, 1 - abs(position))
                    val vertMargin = pageHeight * (1 - scaleFactor) / 2
                    val horzMargin = pageWidth * (1 - scaleFactor) / 2
                    if (position < 0) {
                        page.translationX = horzMargin - vertMargin / 2
                    } else {
                        page.translationX = -horzMargin + vertMargin / 2
                    }
                    page.scaleX = scaleFactor
                    page.scaleY = scaleFactor
                    page.alpha = MIN_ALPHA +
                            (scaleFactor - MIN_SCALE) /
                            (1 - MIN_SCALE) * (1 - MIN_ALPHA)
                }
                else -> {
                    page.alpha = 0f
                }
            }
        }
    }

    companion object {
        private const val MIN_SCALE = 0.85f
        private const val MIN_ALPHA = 0.5f
    }
}